### PR TITLE
feat: Decorate the Nameable class

### PR DESCRIPTION
**Pull request by gvgesanto2**
## Description

**Task:** https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_decorator_v2.md

This PR makes the following changes:
- Creates the ``Nameable`` class, which the ``Person`` class inherits from.
- Creates a base ``Decorator`` class, which inherits from ``Nameable``.
- Creates the ``CapitalizeDecorator`` and ``TrimmerDecorator`` classes which inherit from ``Decorator``.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [X] My code follows the style guidelines of this project (no linters errors)
- [X] I have checked my code and corrected any misspellings
- [X] I have deleted any unused code as comments in my codebase

### CODE REVIEW

File: decorators/trimmer_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require_relative 'decorator'
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.strip
+  end
+end
```
Comment (rloterh) [inline decorators/trimmer_decorator.rb]: - [ ] Your Trimmer Decorator class is not working as intended. Trimming the string ` maximilianus` should return a string of length _10_. Kindly improve your `TrimmerDecorator` class by adding an appropraite method to trim a given string. This should be similar to:

```ruby
class TrimmerDecorator < Decorator
    # The name has a maximum of 10 characters. If it's longer it should trim the word.
    def correct_name
      @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
    end
end
```
**Running your app:**
image

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Gabriel, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Highlights 👍  
- Correctly implemented _Capital Decorator_


## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


**You can also consider:**
- Kindly add `Getting Started` section to your ReadMe
- I think you should change the header title _Authors_ to _Author_ in your ReadMe, as this project is worked on by sole contributor. 
- MIT license seems inappropriate, kindly fix it.
Implementing the above changes will make your ReadMe appear more professional.


<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gvgesanto2** [comment]
## Description

**Task:** https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_decorator_v2.md

This PR addresses the changes requested by the [last review](https://github.com/gvgesanto2/microverse-school-library/pull/2#discussion_r960063678).

**Review by ShahierNashaat — APPROVED** [review]
Hi @gvgesanto2, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Your code is clean and well documented ✔️

# Approved :ok_hand: :ok_hand: :ok_hand:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**